packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1662,25 +1662,25 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       ),
     );
 
-    final Widget prefix = decoration.prefixText == null ? null :
-      new AnimatedOpacity(
+    final Widget prefix = decoration.prefix == null && decoration.prefixText == null ? null
+        : new AnimatedOpacity(
```

Comment:
It would be nice to honor the prefix textStyle here. I'd suggest defining a trivial private widget like _AffixText that deals with suffix and prefix and their AnimatedOpacity:
```
new _AffixText(
  text: prefixText, 
  chlid: prefix, 
  style: decoration.prefixStyle ?? hintStyle,
  labelIsFloating: widget._labelIsFloating,
)
```

Where `_AffixText` either builds a DefaultTextStyle that contains `child` or  a new Text widget with the specified style.